### ISSUE TITLE
“Your keystore is corrupted”

### ISSUE THREAD

**Issue by axelsimon** [guardianproject/ChatSecureAndroid#700]
Hi,

Out of the blue, I tot this message this morning following a ChatSecure crash:
>Your keystore is corrupted.
Please re-install ChatSecure or 'clear all data' for the app

Trying to enter any current conversation crashes ChatSecure.
Trying to a new conversation crashes ChatSecure.

This is pretty bad, especially considering that, as a user, I don't feel I've “done anything” to deserve this, if you know what I mean :)

Re-installing or clearing ChatSecure's data means I'll lose any information stored in the local history, including pictures which may only exist in this history (as they haven't been exported to the external (insecure by default) SD card).

I'm in a situation where I can afford to lose that data, but others may not be, and on the field, this could prove to be a very nasty bug for certain people.

I'm happy to provide logs to help solve this.

Cheers,

axel

**Comment by axelsimon** [comment]
Uh, sorry, just saw the big warning to report bugs elsewhere, on https://dev.guardianproject.info/projects/chatsecure/issues
Shall do.

**Comment by n8fr8** [comment]
Yes keystore corruption is definitely a bug we have worked and are still making sure it doesn't happen. 

What version are you using? From Google Play or elsewhere?

(thanks for posting to our tracker, but thought i would still reply here since you have a very important issue to address)

**Comment by axelsimon** [comment]
https://dev.guardianproject.info/issues/5573

I added that info to your redmine, ChatSecure 14.2.0, installed via F-droid + Guardian Project repo, on Android 5.0.2 (rooted). Let me know if I can help in other ways!

Thanks for answering very quickly too :)
